### ISSUE TITLE
v4.0.29.1 feeds implementation

### ISSUE THREAD

**Issue by idekaresearch** [wet-boew/wet-boew#8541]
I am implementing v4.0.29.1 and cannot get the Feeds to work.

I still see references to the Yahoo API when I load my page. 

1) Is there still a Yahoo API reference?

2) Is it mandatory to place the data-ajax attribute **before** the href attribute [ I ask because our TinyMCE editor reverses the order of these when it cleans up the mark-up ]

3) Can the data-ajax reference a .txt file instead of a .json file? [ I ask because .json files are not automatically served up by IIS without configuration changes ]

I downloaded v4.0.29 of the Canada.ca (GCWeb) theme and replaced the 2 JS files with those from v4.0.29.1 Web Experience Toolkit (WET) theme. 
Is this correct?

**Comment by duboisp** [comment]
> I am implementing v4.0.29.1 and cannot get the Feeds to work.

Is your JSON atom-like feed are properly structured?
* Documentation: https://wet-boew.github.io/wet-boew/docs/ref/feeds/feeds-en.html#json-atom-like-feed
* Example: https://github.com/wet-boew/wet-boew/blob/master/src/plugins/feeds/demo/manitoba-en.atom.json

Do you have a code sample where I try your implementation?

> Is there still a Yahoo API reference?

Yes Yahoo API reference call wasn't removed. We are going to remove the Yahoo API reference in the next version of WET-BOEW v4.0.30. Although, if the feed plugin the ```data-ajax``` attribute is defined in the anchor, then it will ignore any value set to the ```href``` to fetch the feed.

> Is it mandatory to place the data-ajax attribute before the href attribute [ I ask because our TinyMCE editor reverses the order of these when it cleans up the mark-up ]

No, the attribute order don't matter in HTML.

> Can the data-ajax reference a .txt file instead of a .json file? [ I ask because .json files are not automatically served up by IIS without configuration changes ]

I tried with my local apache server with a .txt file in Chrome and it worked fine.

> I downloaded v4.0.29 of the Canada.ca (GCWeb) theme and replaced the 2 JS files with those from v4.0.29.1 Web Experience Toolkit (WET) theme.
Is this correct?

This should work or as I alternative your can also download the [v4.0.29.1 full build](https://github.com/wet-boew/wet-boew/releases/tag/v4.0.29.1) and then replace your local files:  wet-boew.min.js and wet-boew.js

**Comment by idekaresearch** [comment]
Thanks for the response.
It's working now.

Does the Feeds widget only now work with a JSON format?

If so, is it possible for the Feeds widget to just parse the ATOM xml file, and internally convert this to JSON in the WET framework? A future build perhaps?

Our web content management system, like many have been coded to push out XML for the ATOM feeds, in an automated system with web page publishing. Our back-end code must now be updated to push out a duplicate JSON format for this important widget to work.

For now it will be a burden for our clients; every time they publish specific pages, they will have to open the XML file, pass it through an XML-JSON converter, upload and publish the JSON file.

**Comment by duboisp** [comment]
> Does the Feeds widget only now work with a JSON format?

Yes that is true after this patch has been applied: https://github.com/wet-boew/wet-boew/releases/tag/patch-v4.0.21-to-v4.0.29

> If so, is it possible for the Feeds widget to just parse the ATOM xml file, and internally convert this to JSON in the WET framework? A future build perhaps?

Yes, the upcoming version of WET-BOEW (4.0.30) would query directly the XML file.  PR #8534 is about that. However, that type of fetch might require you to configure the CORS HTTP header on the server.
